Web/Scotty/Internal/Types.hs

```diff
@@ -74,9 +74,10 @@ addRoute r s@(ScottyState {routes = rs}) = s { routes = r:rs }
 addHandler :: ErrorHandler e m -> ScottyState e m -> ScottyState e m
 addHandler h s = s { handler = h }
 
-newtype ScottyT e m a = ScottyT { runS :: State (ScottyState e m) a }
-    deriving ( Functor, Applicative, Monad )
+newtype ScottyT' e m n a = ScottyT { runS :: StateT (ScottyState e m) n a }
```

Comment:
Under what circumstances would you want to use `ScottyT` versus `ScottyT'`? This would be a good thing to provide Haddocks for.